### PR TITLE
friends: bug-fixes

### PR DESCRIPTION
**Pull request by austinhallock**
- load spinner for request avatar, join, login, forgot password, reset password
- changed sign in button to not be raised
- rm focus state on links (dotted border)
- have phoneservice use digit number instead of what user entered if ipinfo fails

### CODE REVIEW

File: src/components/forgot_password/index.coffee [hunk 1]
```diff
@@ -34,18 +39,22 @@ module.exports = class ForgotPassword
...
   render: =>
-    {$phoneNumberInput, $signinButton} = @state()
+    {$phoneNumberInput, $signinButton, $spinner, isLoading} = @state()
 
     z '.z-forgot-password',
       z 'form.form', {
         onsubmit: (e) =>
           e.preventDefault()
-          @recover().catch log.trace
+          @recover().catch (err) =>
```
Comment (Zolmeister) [inline src/components/forgot_password/index.coffee]: double catch

File: src/components/forgot_password/index.coffee [hunk 2]
```diff
@@ -56,9 +65,14 @@ module.exports = class ForgotPassword
...
-              @recover().catch log.trace
+        if isLoading
+          $spinner
+        else
+          z 'div.reset-button',
+            z $signinButton,
+              text: 'Reset'
+              onclick: (e) =>
+                e.preventDefault()
+                @recover().catch (err) =>
```
Comment (Zolmeister) [inline src/components/forgot_password/index.coffee]: ^

File: src/components/forgot_password/index.coffee (hunk 1 above)
Comment (Zolmeister) [inline src/components/forgot_password/index.coffee]: *state set

File: src/components/join/index.coffee [hunk 3]
```diff
@@ -71,16 +78,21 @@ module.exports = class Join
...
+      $spinner, isLoading} = @state()
 
     z '.z-join',
       z 'form.form', {
         onsubmit: (e) =>
           e.preventDefault()
-          @signup(fromUserId).catch log.trace
+          @signup(fromUserId).catch (err) =>
+            log.trace err
+            @state.set isLoading: false
```
Comment (Zolmeister) [inline src/components/join/index.coffee]: double state set

File: src/components/login/index.coffee [hunk 4]
```diff
@@ -45,20 +50,24 @@ module.exports = class Login
...
+      $forgotButton, $spinner, isLoading} = @state()
 
     z '.z-login',
       z 'form.form', {
         onsubmit: (e) =>
           e.preventDefault()
-          @login().catch log.trace
+          @login().catch (err) =>
+            log.trace err
+            @state.set isLoading: false
```
Comment (Zolmeister) [inline src/components/login/index.coffee]: again

File: src/components/reset_password/index.coffee [hunk 5]
```diff
@@ -58,19 +66,23 @@ module.exports = class ResetPassword
...
+      $resendButton, $spinner, isLoading} = @state()
 
     z '.z-reset-password',
       z 'form.form', {
         onsubmit: (e) =>
           e.preventDefault()
-          @changePassword(phone).catch log.trace
+          @changePassword(phone).catch (err) =>
+            log.trace err
+            @state.set isLoading: false
```
Comment (Zolmeister) [inline src/components/reset_password/index.coffee]: ^

File: src/components/request_avatar_card/index.coffee [hunk 6]
```diff
@@ -13,7 +14,7 @@ dataUriToBlob = (dataUri) ->
   binary = atob(dataUri.split(',')[1])
   byteArray = _.map _.range(binary.length), (i) ->
     binary.charCodeAt(i)
-  mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0]
+  mimeString = dataUri.split(',')[0].split(':')[1].split(';')[0]
```
Comment (Zolmeister) [inline src/components/request_avatar_card/index.coffee]: tests

File: src/services/phone.coffee [hunk 7]
```diff
@@ -23,6 +23,7 @@ class PhoneService
       .then (locationData) ->
         internationalPhone = dialCodes[locationData.country] + digitPhone
         if internationalPhone.match new RegExp INTERNATIONAL_PHONE_REGEX
+          log.trace new Error 'invalid phone number'
```
Comment (Zolmeister) [inline src/services/phone.coffee]: er... Isn't this suppose to go in the else? Also throw the phone number in there

**Comment by Zolmeister** [comment]
rebase

**Comment by austinhallock** [comment]
done, we'll get to the type: 'submit' stuff later

**Pull request merged**